WarhammerOldWorld/WarhammerOldWorld/Modules/TroopAttributes/TroopAttributeSubModule.cs

```diff
@@ -15,33 +15,36 @@ protected internal override void OnSubModuleLoad()
 
         public override void OnMissionBehaviourInitialize(Mission mission)
         {
-            mission.AddMissionBehaviour(GetTroopAttributeMissionLogic());
+            mission.AddMissionBehaviour(new TroopAttributeMissionLogic());
+            PopulateTroopAttributes();
 
             // Replace the default morale interaction logic with our new custom morale logic
             mission.RemoveMissionBehaviour(mission.GetMissionBehaviour<AgentMoraleInteractionLogic>());
-            mission.AddMissionBehaviour(new CustomBaseAgentMoraleInteractionLogic());
+            mission.AddMissionBehaviour(new TowAgentMoraleInteractionLogic());
         }
 
-        private TroopAttributeMissionLogic GetTroopAttributeMissionLogic()
+        private void PopulateTroopAttributes()
```

Comment:
This method needs to be removed eventually and any callers need to use the relevant ObjectManager instead. As far as I know, there's no current structure for how we want to manage our XML with ObjectManagers (for example, currently, the XMLs are split by culture. Do we want an object manager for each culture, i.e. VampireCountsObjectManager?) Need to make a ticket to come up with a design for this. 

Maybe we don't even want to use the current ObjectManager idea that's in the code and we can remove that entirely. I'll leave that decision for some other place instead of in this commit comment.